UnitySDK/Assets/ML-Agents/Scripts/LearningBrain.cs

```diff
@@ -27,460 +21,247 @@ namespace MLAgents
     [CreateAssetMenu(fileName = "NewLearningBrain", menuName = "ML-Agents/Learning Brain")]
     public class LearningBrain : Brain
     {
-        [System.Serializable]
-        private struct TensorFlowAgentPlaceholder
-        {
-            public enum TensorType
-            {
-                Integer,
-                FloatingPoint
-            };
-
-            public string name;
-            public TensorType valueType;
-            public float minValue;
-            public float maxValue;
-        }
+        private const long ApiVersion = 1; 
+        private List<string> _failedModelChecks = new List<string>();
+        private long _modelMemorySize;
+        
+        private TensorGeneratorInvoker _tensorGeneratorInvoker;
+        private TensorApplierInvoker  _outputTensorApplierInvoker;
+        
+        public Model model;
```

Comment:
why are `model` and `inferenceDevice` public?

Reply:
They are an asset and enum respectively set in the UI by the user.

Reply:
@vladimiroster thanks for the clarification. @vincentpierre - can you expand the class description to include this.

Reply:
I will make a note on `model` and replace `SetModel(Model)` with `LoadModel()`, but I think inferenceDevice should be hidden given that it is useless for TF#.

Reply:
If it's invisible, how will the users select it on the other thing or the other-other thing?